pkg/controller/ingress/concurrency_watchdog.go

```diff
@@ -219,7 +220,16 @@ func (c *ConcurrencyWatchdog) tick(ctx context.Context) error {
 			ctx := logr.NewContext(ctx, lgr)
 			count, err := target.ScrapeFn(ctx, c.restClient, &pod)
 			if err != nil {
-				lgr.Error(err, "scraping pod", "name", pod.Name)
+
+				// check if pod is still ready. Pod might have become unready after checking it (this solves a race condition).
+				// we ignore an error on the podIsActive call, we want the retErr to be the error from scrapping not from checking if
+				// the pod is active.
+				if active, err := podIsActive(ctx, lgr, c.client, client.ObjectKeyFromObject(&pod)); err == nil && !active {
```

Comment:
do we want to propagate this error at all? I think it could be helpful to log it -- could help explain why metrics scraping failed going forward

Reply:
like a lgr.Error("failed to check if pod is active:" , err)

Reply:
pushed an update to log in podIsActive